### PR TITLE
Solutions to Lecture 2

### PR DESCRIPTION
**Pull request by duobei**
### Solutions for Lecture 2

cc @chshersh @vrom911

### CODE REVIEW

File: src/Lecture2.hs [hunk 1]
```diff
@@ -97,7 +113,9 @@ spaces.
 
 🕯 HINT: look into Data.Char and Prelude modules for functions you may use.
 -}
-dropSpaces = error "TODO"
+
+dropSpaces :: [Char] -> [Char]
+dropSpaces = filter (/= ' ')
```
Comment (chshersh) [inline src/Lecture2.hs]: This is a nice and short implementation 👍🏻 

Unfortunately, it doesn't work on infinite lists 😞 

If I generate a list like this: `myList = "foo" ++ repeat ' '` and do `take 5 (dropSpaces myList)`, the function will hang. However, it's possible to implement this function so the above won't hang.

File: src/Lecture2.hs [hunk 2]
```diff
@@ -191,7 +215,11 @@ verify that.
 [1,2,3,4,7]
 -}
 merge :: [Int] -> [Int] -> [Int]
-merge = error "TODO"
+merge [] xs = xs
+merge xs [] = xs
+merge (x:xs) (y:ys)
+  | x > y = y : merge (x:xs) ys
```
Comment (chshersh) [inline src/Lecture2.hs]: You can implement this function in a slightly more efficient way by having a separate case when both elements in the heads are equal. With this approach, you'll be able to shorten lists faster 🐴

**Review by chshersh — APPROVED** [review]
Great solutions! 👏🏻 
I feel like you knew Haskell quite well before this course 😅 

I've added a few suggestions to your solutions. But the rest looks great 💯

File: src/Lecture2.hs (hunk 1 above)
Comment (duobei) [reply src/Lecture2.hs]: Thanks. I have fixed it.

**Pull request merged**